### PR TITLE
Integrate SpanMarker library

### PR DESCRIPTION
**Pull request by tomaarsen**
Hello!

## Pull Request overview
* Integrate with the [SpanMarker](https://github.com/tomaarsen/SpanMarkerNER) library for Named Entity Recognition.

## Details
[SpanMarker](https://github.com/tomaarsen/SpanMarkerNER) is a model architecture for Named Entity Recognition that is tightly implemented on top of transformers. Consequently, loading a SpanMarker model, e.g. [`tomaarsen/span-marker-bert-base-fewnerd-fine-super`](https://huggingface.co/tomaarsen/span-marker-bert-base-fewnerd-fine-super) is as simple as:
```python
from span_marker import SpanMarkerModel

model = SpanMarkerModel.from_pretrained("tomaarsen/span-marker-bert-base-fewnerd-fine-super")
```

Feel free to try out this [Space](https://huggingface.co/spaces/tomaarsen/span-marker-bert-base-fewnerd-fine-super) to get a feel for the power of the architecture.

I've followed the documentation [here](https://huggingface.co/docs/hub/models-adding-libraries), but it wasn't extremely clear on which sections should be added for a succesful integration, so please let me know if I forgot any parts.

This PR is related to https://github.com/huggingface/api-inference-community/pull/225, which adds support for SpanMarker models to API Inference endpoints.

- Tom Aarsen

### CODE REVIEW

**Comment by tomaarsen** [comment]
On a slightly related note - once I know that this is indeed all I have to update to integrate a library, I can certainly integrate 🤗 [SetFit](https://github.com/huggingface/setfit) fully.

**Review by osanseviero — COMMENTED** [review]
Thanks for the contribution! In general, we only merge new library PRs once we have a good number of models on the Hub, otherwise, we would have a filter button that would lead to 4 models at the moment. That said, we discussed with the team and would be ok moving this PR forward :fire: Let's aim to grow this significantly and gets lots of models! :partying_face: 

For completeness, we would also need to update some other sections
* Add some docs such as https://github.com/huggingface/hub-docs/blob/main/docs/hub/speechbrain.md with a brief intro (and update https://github.com/huggingface/hub-docs/blob/main/docs/hub/_toctree.yml to make sure this is included
* Add library to list of tasks https://github.com/huggingface/hub-docs/blob/main/tasks/src/const.ts#L43  
* Add library to following table https://github.com/huggingface/hub-docs/blob/main/docs/hub/models-libraries.md?plain=1#L30

**Comment by tomaarsen** [comment]
@osanseviero 
That is wonderful news! I'm very excited to hear it. SpanMarker is still young, and I'm still experimenting with some ideas that could result in non-backwards compatible changes, so I haven't written about the library elsewhere yet. That is why there's only 4 models at the moment - I'll be sure to push for more (i.e. ones that aren't mine 😉) once I'm ready to start advertising the library.

On that topic, I was wondering how you handle versioning? I intend to release v1.0.0 this month, which may or may not be backwards compatible with v0.X.X models. Currently, most v0.X.X models are my own, so I'm not completely sure whether I should messy up my codebase for backwards compatibility. May I assume that the version used for e.g. Inference Endpoints is always the most recent version?

I'll add the other changes that you mention to this PR. Then, perhaps it is best if I convert this PR to a draft until v1.0.0 later this month, especially if the implementation of backwards compatibility for v0.X.X to v1.0.0 adds too much bloat.

- Tom Aarsen

**Comment by osanseviero** [comment]
> On that topic, I was wondering how you handle versioning? I intend to release v1.0.0 this month, which may or may not be backwards compatible with v0.X.X models. Currently, most v0.X.X models are my own, so I'm not completely sure whether I should messy up my codebase for backwards compatibility. May I assume that the version used for e.g. Inference Endpoints is always the most recent version?

Ideally the library should maintain backwards compatibility, but it might be ok to break things if the library is in a very early stage and usage is not large yet. When we do backwards compatible changes, we go through a deprecation period so people get warnings and we provide proper time to update code. As for Inference API, it will use the version pinned in the `requirements.txt` file, so it depends on that

File: docs/hub/models-libraries.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ The table below summarizes the supported libraries and their level of integratio
 | [Sample Factory](https://github.com/alex-petrenko/sample-factory)    | Codebase for high throughput asynchronous reinforcement learning.          | ❌ | ✅ | ✅ | ✅ |
 | [Sentence Transformers](https://github.com/UKPLab/sentence-transformers)    | Compute dense vector representations for sentences, paragraphs, and images.          | ✅ | ✅ | ✅ | ✅ |
 | [spaCy](https://github.com/explosion/spaCy)                                 | Advanced Natural Language Processing in Python and Cython.                           | ✅ | ✅ | ✅ | ✅ |
+| [SpanMarker](https://github.com/tomaarsen/SpanMarkerNER)                    | Familiar, simple and state-of-the-art Named Entity Recognition.                      | ✅ | ✅ | ✅ | ✅ |
```
Comment (osanseviero) [inline docs/hub/models-libraries.md:29]: Is there a push to hub mechanism?

**Review by osanseviero — APPROVED** [review]
Nice! Thanks a lot! I left a question and then we can merge

File: docs/hub/models-libraries.md (hunk 1 above)
Comment (tomaarsen) [reply docs/hub/models-libraries.md:29]: There sure is! `SpanMarkerModel` subclasses `PreTrainedModel`, so it's as simple as `model.push_to_hub`.

**Comment by osanseviero** [comment]
Up to you! I think it makes sense until official release before merging this

**Comment by tomaarsen** [comment]
Agreed. I'll draft this and I'll report back when it's ready! 
Same for https://github.com/huggingface/api-inference-community/pull/225.

Thanks for the help so far :heart:

**Comment by tomaarsen** [comment]
[SpanMarker](https://github.com/tomaarsen/SpanMarkerNER) just released with v1.0.0, so I'm ready to get this integration merged!

**Review by osanseviero — APPROVED** [review]
Very cool! :fire:

**Pull request merged**